proxy/src/main/java/com/cpsc559/proxy/controller/ProxyController.java

```diff
@@ -0,0 +1,95 @@
+package com.cpsc559.proxy.controller;
+
+import jakarta.servlet.http.HttpServletRequest;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.http.*;
+import org.springframework.web.bind.annotation.*;
+import org.springframework.web.reactive.function.client.WebClient;
+import reactor.core.publisher.Flux;
+import reactor.core.publisher.Mono;
+
+import java.util.Arrays;
+import java.util.List;
+
+@RestController
+public class ProxyController {
+
+    private final WebClient webClient;
+
+    // Base URL for the primary backend server -> defined in application.properties
+    @Value("${backend.primary.url}")
+    private String primaryBackendUrl;
+
+    // List of comma-seperated URLs for the backup servers -> defined in application.properties
+    @Value("${backend.backups.urls}")
+    private String[] backupBackendUrls;
+
+    public ProxyController(WebClient webClient) {
+        this.webClient = webClient;
+    }
+
+    // All requests go through here
+    @RequestMapping("/**")
```

Comment:
Tried running the application with the proxy and ran into CORS errors. This should resolve those:
```suggestion
    @CrossOrigin(origins = "*", methods = {RequestMethod.GET, RequestMethod.POST, RequestMethod.PUT, RequestMethod.DELETE, RequestMethod.OPTIONS})
    @RequestMapping("/**")
```